### PR TITLE
Add file CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by inknos**
This commit adds the new file CONTRIBUTING.md with a detailed guide on how to contribute to the codebase. It also removes the duplicate information from README.md and updates the documentation on readthedocs. The section "contributing" is now "best practices" to avoid confusion.

For better readability, this is how the updated files should look.
[README.md](https://github.com/inknos/dnf5/blob/branch-CONTRIBUTING.md/README.md)
[CONTRIBUTING.md](https://github.com/inknos/dnf5/blob/branch-CONTRIBUTING.md/CONTRIBUTING.md)

### CODE REVIEW

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:22]:
```suggestion
* Problems found during a new developer environment setup.
* Gaps in our Quickstart Guide or documentation.
* Bugs in our automation scripts.
```

File: CONTRIBUTING.md
Comment (jan-kolarik) [inline CONTRIBUTING.md:145]: I'd suggest to also cover added functionality with unit tests. Something like this?

### Code should be tested

If your pull request includes new or modified functionality within the library, we kindly request that you provide matching unit tests in the project's `test` directory to cover these changes. However, if the changes only affect the command-line interface, you may provide related CI tests in the [ci-dnf-stack](https://github.com/rpm-software-management/ci-dnf-stack) component. If you need assistance with this, please feel free to ask the maintainers for help.

File: README.md
Comment (jan-kolarik) [inline README.md:40]: Minor: maybe I would drop the `F` in the Fedora column values and have just `36`, `37` and `38`?

**Comment by jan-kolarik** [comment]
In overall, it looks very good to me! It's a great step forward compared to the previous version.

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:104]:
```suggestion
To run the tests, follow the steps to build the code and then run:
```

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:111]:
```suggestion
As an alternative, tests can be executed in a verbose mode:
```

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:120]:
```suggestion
To install the built package, we recommend building an rpm package.
```

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:122]:
```suggestion
To build rpms using tito, run:
```

File: CONTRIBUTING.md
Comment (inknos) [reply CONTRIBUTING.md:145]: I added the section, will push an updated version in a moment

**Comment by inknos** [comment]
I am updating the PR locally and marking the conversations as resolved once I revise the mistakes in my local clone.

File: CONTRIBUTING.md
Comment (jan-kolarik) [inline CONTRIBUTING.md:145]:
```suggestion
Otherwise, please, [test it locally](#development-environment-setup).
```

File: CONTRIBUTING.md
Comment (jan-kolarik) [inline CONTRIBUTING.md:153]:
```suggestion
If you need assistance with this, please feel free to ask the maintainers for help.
```

File: CONTRIBUTING.md
Comment (jan-kolarik) [inline CONTRIBUTING.md:68]:
```suggestion
  - Set the [labels](https://github.com/rpm-software-management/dnf5/labels) accordingly.
```

File: CONTRIBUTING.md
Comment (jan-kolarik) [inline CONTRIBUTING.md:73]:
```suggestion
    [ready for review](https://github.com/rpm-software-management/dnf5/labels/ready%20for%20review).
```

**Review by jan-kolarik — APPROVED** [review]
There's some minors like using dots consistently, etc., but we don't need to go into that nit-picking...

File: CONTRIBUTING.md
Comment (jan-kolarik) [inline CONTRIBUTING.md:172]: Another thing occurred to me in this context. Maybe we could add a note about adding a proper description of the changes in the commit message and PR description?

Comment (inknos) [reply CONTRIBUTING.md:172]: oh we could actually setup the commit template. what do you think?
https://github.com/rpm-software-management/dnf/blob/master/.git-commit-template

Comment (jan-kolarik) [reply CONTRIBUTING.md:172]: But you still need to setup these template manually, right? Because I don't see the template by default using `git commit`. And also it seems not working properly in some IDEs like VS code from the link given.

If it couldn't be automated, I am rather for documenting the suggested format there.

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:18]:
```suggestion
We welcome your feedback on different areas of our project including the following:
```

Just trying to make it a bit more simple.

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:21]:
```suggestion
* Gaps in our Quickstart Guide or other documentation.
```
If that makes sense.

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:27]:
```suggestion
If you are experiencing problems with our project, let us know by opening an 
[issue](https://github.com/rpm-software-management/dnf5/issues/new) and let us know!
If you want to file a bug on Fedora, you can open the bug in
[Bugzilla](https://bugzilla.redhat.com/enter_bug.cgi?product=Fedora&component=dnf5).
```

Not sure if all my suggestions make sense here, though :-)  Other option is to replace "with our project" with "the DNF5 package manager".

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:45]:
```suggestion
If you are a new contributor, you can search for issues with the [_good first issue_](https://github.com/rpm-software-management/dnf5/labels/good%20first%20issue) label. These issues are suitable for contributors who are not core maintainers and want to get familiar with the codebase.

If you are a regular contributor, you can search for issues with the _help wanted_ label.
```

Here, I'm not sure if I interpreted the info in the correct way :-D Also, would it be possible to add also the link to the _help wanted_ issues search result similarly to what you have for _good first issue_?

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:50]:
```suggestion
Note that sometimes there can be no issues with the _good first issue_ or _help wanted_ label. Do not worry! There is likely still something for you to work on. If you want to contribute but you
do not know where to start or cannot find a suitable issue, ask for help by reaching out on one of our
[channels](#ask-for-help).
```

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:54]:
```suggestion
If there is an issue that you would like like to work on, add a comment saying
that you want to work on this issue, for example, "I want to work on this", and assign the issue to yourself.
```

Not sure about the "assign the issue to yourself"..Does that sound correct?

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:79]:
```suggestion
Once you have a contribution ready, complete the following steps:
- Submit a pull request (PR).
  - Set the necessary [labels](https://github.com/rpm-software-management/dnf5/labels).
  - If you need help while working on the PR, set the
    [_work in progess_](https://github.com/rpm-software-management/dnf5/labels/work%20in%20progress) label
    and wait for the PR to be triaged.
  - If you are ready for a code review by one of the maintainers, set the
    [_ready for review_](https://github.com/rpm-software-management/dnf5/labels/ready%20for%20review) label.
    Triaging usually happens on Mondays or Thursdays.
- Review the PR.
  - Discuss the PR with one of the maintainers. Note that some changes might be requested.
  - Review again.
- Merge the PR.
```

I would also use the numbered list for the main steps. This ensures that folks go step by step and do not perform these steps in a different order (even though it is obvious that those steps should be performed one by one :-D ) But this is just a suggestion :-)

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:82]:
```suggestion
To setup a development environment, complete the following steps:
```

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:140]:
```suggestion
When you submit your pull request or push new commits to it, our automated
systems will run some checks on your new code. We require that your pull request
passes these checks. However, note that there are more criteria that your pull request must pass before we can
accept and merge it. We recommend that you ensure the following locally
before you submit your code:
```

Comment (mpershina) [reply CONTRIBUTING.md:140]: I would also turn the following two criteria into the bulleted list as follows:

- Code builds.
  Unless you 
  [Test your code locally](#development-environment-setup) unless you need our help or want us to review
  your _work in progress_ changes.
- Code is tested.
  If your pull request includes a new or modified functionality within the library, provide
  matching unit tests in the project's test directory to cover the changes. However, if the changes only affect
  the command-line interface, you can provide related CI tests in the
  [ci-dnf-stack](https://github.com/rpm-software-management/ci-dnf-stack) component.
  If you need our assistance, ask the maintainers for help.
- Code passed sanity checks.

And so on. Does that sound ok?

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:158]:
```suggestion
In the dnf5` directory, run:
```

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:162]:
```suggestion
git add _<yourchanges>_
git commit -m "_<your commit>_"
```

This is how we usually express the replaceable values.

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:168]:
```suggestion
All these checks will run as part of a PR action on GitHub. Therefore, make sure not to skip the checks if you do not want to amend your contribution.
```

File: CONTRIBUTING.md
Comment (inknos) [reply CONTRIBUTING.md:172]: I created an issue to track this point. I believe there is a good margin of improvement and automation if we think this right.

https://github.com/rpm-software-management/dnf5/issues/498

File: CONTRIBUTING.md
Comment (inknos) [reply CONTRIBUTING.md:45]: you are correct; however, we have a change in the workflow now, so I will simply remove the last line "If you are a regular contributor... wanted label." since this piece of information is not valid anymore

Comment (mpershina) [reply CONTRIBUTING.md:45]: Oki, got it :-)

File: CONTRIBUTING.md
Comment (inknos) [reply CONTRIBUTING.md:79]: I like the suggestion, I'll make the change

File: CONTRIBUTING.md
Comment (inknos) [reply CONTRIBUTING.md:162]: not sure that the italic would format inside the code block but i'll try

File: CONTRIBUTING.md
Comment (inknos) [reply CONTRIBUTING.md:140]: sounds good

**Comment by inknos** [comment]
I made another round of changes and the pr is now updated with a commit "Changes requested" to highlight the corrections that were made. Once the PR is approved I will squash all the commits into one

**Comment by inknos** [comment]
@mpershina do you think another round of review would be necessary before merging?

File: CONTRIBUTING.md
Comment (jan-kolarik) [inline CONTRIBUTING.md:9]: These links under #development-environment-setup section seem not to lead anywhere ...

File: README.md
Comment (mpershina) [inline README.md:31]:
```suggestion
 * C - not implemented, not a priority for any of our existing API users
```

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:115]:
```suggestion
   - Build the rpm by using tito:

     ```
     tito build --rpm --test
     ```

   - Build the rpm from git:
```

File: CONTRIBUTING.md
Comment (mpershina) [inline CONTRIBUTING.md:143]:
```suggestion
* Code must pass sanity checks.
```

File: README.md
Comment (mpershina) [inline README.md:31]:
```suggestion
Note, however, that DNF5 cannot yet interface with the following programming languages:
```

File: CONTRIBUTING.md
Comment (inknos) [reply CONTRIBUTING.md:115]: it's actually more than one rpm being built

File: CONTRIBUTING.md
Comment (jan-kolarik) [inline CONTRIBUTING.md:15]: What is the `Quickstart Guide`? Is it referring to this docs or something else?

Comment (inknos) [reply CONTRIBUTING.md:15]: it's inaccurate, right, it comes from the template I used. I would change it to "Gaps in our documentation"

File: CONTRIBUTING.md
Comment (jan-kolarik) [inline CONTRIBUTING.md:133]: Doubled dot: `..` -> `.`

**Comment by jan-kolarik** [comment]
Nice work! What an effort! Thank you, @inknos and @mpershina, for your great contributions.

**Pull request merged**
